src/main/java/TicTacToe/WinLogic.java

```diff
@@ -0,0 +1,91 @@
+package TicTacToe;
+
+public class WinLogic {
+    private Board board;
+    private boolean win = false;
+    private boolean gameContinues = true;
+
+    public WinLogic(Board board) {
+        this.board = board;
+    }
+
+    public WinLogic() {
+    }
+
+    public GameStatus checkWin() {
+        for (char symbol : new char[]{board.CIRCLE, board.CROSS}) {
+            for (int i = 0; i < board.board.length; i++) {
+                if (checkSameSymbolInColumn(i, symbol) || checkSameSymbolInRow(i, symbol) ||
+                        checkSameSymbolInMainDiagonal(symbol) || checkSameSymbolInSecondaryDiagonal(symbol)) {
+                    win = true;
+                    gameContinues = false;
+                    return symbol == board.CIRCLE ? GameStatus.O_WON : GameStatus.X_WON;
```

Comment:
Zmienne `win` i `gameContinues` są niepotrzebene. To klient powinnien interpretować statusy gry na 'win' i `gameContinues`. WinLogic ma zwracać stauts gry i tyle.